packages/outline-view/src/browser/outline-view-contribution.ts

```diff
@@ -40,4 +51,42 @@ export class OutlineViewContribution extends AbstractViewContribution<OutlineVie
     async initializeLayout(app: FrontendApplication): Promise<void> {
         await this.openView();
     }
+
+    registerCommands(commands: CommandRegistry): void {
+        super.registerCommands(commands);
+        commands.registerCommand(OutlineViewCommands.COLLAPSE_ALL, {
+            isEnabled: widget => this.withWidget(widget, () => true),
+            isVisible: widget => this.withWidget(widget, () => true),
+            execute: () => this.collapseAllItems()
+        });
+    }
+
+    registerToolbarItems(toolbar: TabBarToolbarRegistry): void {
+        toolbar.registerItem({
+            id: OutlineViewCommands.COLLAPSE_ALL.id,
+            command: OutlineViewCommands.COLLAPSE_ALL.id,
+            tooltip: 'Collapse All',
+            priority: 0
+        });
+    }
+
+    /**
+     * Collapse all nodes in the outline view tree.
+     */
+    protected async collapseAllItems(): Promise<void> {
+        const { model } = await this.widget;
+        const root = model.root as CompositeTreeNode;
```

Comment:
How about:
```ts
const root = model.root;
if (CompositeTreeNode.is(root)) {
    model.collapseAll(root);
}
```

please don't use `as`, but do proper type checking

Reply:
Sure, I will fix it ;)

Reply:
It does not seem to be addressed

Reply:
Just addressed in the latest push ;)